src/abstractdataframe/where.jl

```diff
@@ -0,0 +1,191 @@
+
+julia> where(groupby(df, :y), :v => x -> x .> minimum(x), dropmissing=false)
+2×5 DataFrame
+│ Row │ id    │ x    │ y    │ z       │ v     │
+│     │ Int64 │ Bool │ Bool │ Bool?   │ Int64 │
+├─────┼───────┼──────┼──────┼─────────┼───────┤
+│ 1   │ 2     │ 0    │ 1    │ 1       │ 2     │
+│ 2   │ 4     │ 0    │ 0    │ missing │ 12    │
+```
+"""
+function where(df::AbstractDataFrame, @nospecialize(args...);
+               dropmissing::Bool=true, view::Bool=false)
+    row_selector = _get_where_conditions(df, args, dropmissing)
+    return view ? Base.view(df, row_selector, :) : df[row_selector, :]
+end
+
+"""
+    where!(df::AbstractDataFrame, @nospecialize(args...); dropmissing::Bool=true)
+    where!(gdf::GroupedDataFrame{DataFrame}, args...; ungroup::Bool=true, dropmissing::Bool=true)
+
+Update a data frame `df` or a parent of `gdf` in place to contain only rows for
+which all values produced by `args` transformation in a given row is `true`. If
+`dropmissing=true` (the default) returing `missing` in `args` is allowed and
+these rows are dropped. If `dropmissing=false` an error is thrown if `missing`
+is present.
+
+Each argument passed in `args` can be either a single column selector or a
+`source_columns => function` transformation specifier following the rules
+described for [`select`](@ref).
+
+If `ungroup=true` then a `GroupedDataFrame` is returned using the same columns as
+`gdf` to group by.
+
+See also: [`where!`](@ref), [`filter`](@ref), [`filter!`](@ref)
+
+# Examples
+
+```
+julia> df = DataFrame(id=1:4, x=[true, false, true, false], y=[true, true, false, false])
+4×3 DataFrame
+│ Row │ id    │ x    │ y    │
+│     │ Int64 │ Bool │ Bool │
+├─────┼───────┼──────┼──────┤
+│ 1   │ 1     │ 1    │ 1    │
+│ 2   │ 2     │ 0    │ 1    │
+│ 3   │ 3     │ 1    │ 0    │
+│ 4   │ 4     │ 0    │ 0    │
+
+julia> where!(copy(df), :x, :y => ByRow(!));
+
+julia> df
+4×3 DataFrame
+│ Row │ id    │ x    │ y    │
+│     │ Int64 │ Bool │ Bool │
+├─────┼───────┼──────┼──────┤
+│ 1   │ 1     │ 1    │ 1    │
+│ 2   │ 2     │ 0    │ 1    │
+│ 3   │ 3     │ 1    │ 0    │
+│ 4   │ 4     │ 0    │ 0    │
+
+julia> df = DataFrame(id=1:4, y=[true, true, false, false], v=[1, 2, 11, 12]);
+
+julia> where!(groupby(df, :y), :v => x -> x .> minimum(x), dropmissing=false);
+
+julia> df
+2×3 DataFrame
+│ Row │ id    │ y    │ v     │
+│     │ Int64 │ Bool │ Int64 │
+├─────┼───────┼──────┼───────┤
+│ 1   │ 2     │ 1    │ 2     │
+│ 2   │ 4     │ 0    │ 12    │
+```
+"""
+function where!(df::AbstractDataFrame, @nospecialize(args...); dropmissing::Bool=true)
+    row_selector = _get_where_conditions(df, args, dropmissing)
+    delete!(df, findall(!, row_selector))
+    return df
+end
+
+function where(gdf::GroupedDataFrame, @nospecialize(args...);
+               dropmissing::Bool=true, ungroup::Bool=true)
+    row_selector = _get_where_conditions(gdf, args, dropmissing)
+    df = parent(gdf)[row_selector, :]
+    return ungroup ? df : groupby(df, groupcols(gdf))
+end
```

Comment:
I'd put this above with the other `where` method, since the docstrings separate `where` from `where!` quite clearly.

Reply:
moved